crates/sui-types/src/base_types.rs

```diff
@@ -195,10 +196,10 @@ impl From<&PublicKeyBytes> for SuiAddress {
     }
 }
 
-impl From<&PublicKey> for SuiAddress {
-    fn from(pkb: &PublicKey) -> Self {
+impl<T: VerifyingKey> From<&T> for SuiAddress {
+    fn from(pk: &T) -> Self {
         let mut hasher = Sha3_256::default();
-        hasher.update(pkb.as_ref());
+        hasher.update(pk.as_ref());
```

Comment:
Don't we need the flag here (per our discussion)?

Reply:
Yes, we do - sorry about that. I totally forgot this part.